src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -204,6 +204,22 @@ protected ImmutableArray<LocalSymbol> BuildLocals(SyntaxList<StatementSyntax> st
                         {
                             Binder localDeclarationBinder = enclosingBinder.GetBinder(innerStatement) ?? enclosingBinder;
                             var decl = (LocalDeclarationStatementSyntax)innerStatement;
+
+                            decl.Declaration.Type.VisitTypeSyntaxes(type =>
+                            {
+                                if (type is ArrayTypeSyntax arrayTypeSyntax)
+                                {
+                                    foreach (var rankSpecifier in arrayTypeSyntax.RankSpecifiers)
+                                    foreach (var expression in rankSpecifier.Sizes)
+                                    {
+                                        if (expression.Kind() != SyntaxKind.OmittedArraySizeExpression)
+                                        {
+                                            ExpressionVariableFinder.FindExpressionVariables(this, locals, expression, enclosingBinder.GetBinder(expression) ?? enclosingBinder);
```

Comment:
>enclosingBinder.GetBinder(expression) ?? enclosingBinder [](start = 119, length = 56)

Why the binder is different from the one used for the bracketed argument lists and the initializers below?